### PR TITLE
Update tests for KFP v2.

### LINKED ISSUE #254: [Feature] KFP_v2 Step 3 Update the Readme/docs files and tests the pipelines with KFPv2.
### Search before asking

- [X] I searched the [issues](https://github.com/IBM/data-prep-lab/issues) and found no similar issues.

### Component

KFP workflows

### Feature

After splitting the KFP-support libraries, we have to update relevant documentation files and test the pipelines with KFPv2

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by revit13**
/Closes #254

## Why are these changes needed?

Fix tests for kfp v2

### CODE REVIEW

File: kfp/kfp_ray_components/Makefile [hunk 1]
```diff
@@ -51,8 +50,10 @@ reconcile-requirements:
 	sed -i.back "s/kfp-data-processing.*:[0-9].*/$(DOCKER_IMAGE_NAME):${DOCKER_IMAGE_VERSION}/" deleteRayClusterComponent.yaml
 	sed -i.back "s/kfp-data-processing.*:[0-9].*/$(DOCKER_IMAGE_NAME):${DOCKER_IMAGE_VERSION}/" executeRayJobComponent.yaml
 	sed -i.back "s/kfp-data-processing.*:[0-9].*/$(DOCKER_IMAGE_NAME):${DOCKER_IMAGE_VERSION}/" executeRayJobComponent_multi_s3.yaml
+ifneq ($(KFPv2), 1)
 	# TODO remove it for KFPv2
```
Comment (roytman) [inline kfp/kfp_ray_components/Makefile]: we don't need the TODO here

File: transforms/.make.workflows [hunk 2]
```diff
@@ -13,16 +13,30 @@ define set_env_var
 	$(eval export $(1)=$(2))
 endef
 
-# FIXME
+ifeq ($(KFPv2), 1)
```
Comment (roytman) [inline transforms/.make.workflows]: we have the same code snippet in `kfp_ray_components/Makefile`, can they be combined ?

Comment (revit13) [reply transforms/.make.workflows]: Fixed. Thanks

File: kfp/kfp_ray_components/Makefile (hunk 1 above)
Comment (revit13) [reply kfp/kfp_ray_components/Makefile]: Fixed. Thanks

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -57,7 +57,7 @@ jobs:
             - name: Test tools
               run: |
                   make -C tools DOCKER=docker venv test
-    test-kfp-lib:
+    test-kfp-v1-lib:
```
Comment (roytman) [inline .github/workflows/test.yml]: do we need `test-kfp-v2-lib` and `test-kfp-shared-lib`?

File: kfp/kfp_support_lib/kfp_v2_workflow_support/src/workflow_compile_utils/__init__.py [hunk 4]
```diff
@@ -1 +0,0 @@
-from workflow_utils import ONE_HOUR_SEC, ONE_DAY_SEC, ONE_WEEK_SEC, ComponentUtils
```
Comment (roytman) [inline kfp/kfp_support_lib/kfp_v2_workflow_support/src/workflow_compile_utils/__init__.py:1]: should the entire dir be removed?

File: kfp/kfp_support_lib/kfp_v2_workflow_support/src/workflow_support/compile_utils/component.py [hunk 5]
```diff
@@ -58,6 +58,9 @@ def set_s3_env_vars_to_component(
         """
 
         if prefix is not None:
-            for env_name, _ in env2key.items():
-                env2key[prefix + "_" + env_name] = env2key.pop(env_name)
-        kubernetes.use_secret_as_env(task=task, secret_name="s3-secret", secret_key_to_env=env2key)
+            for secret_key, _ in env2key.items():
```
Comment (roytman) [inline kfp/kfp_support_lib/kfp_v2_workflow_support/src/workflow_support/compile_utils/component.py:61]: why do you need this replacement?
if the prefix is None, you will get strange names, started form "_".

File: kfp/kfp_support_lib/kfp_v2_workflow_support/src/workflow_support/pipeline_utils/pipeline_utils.py [hunk 6]
```diff
@@ -72,23 +75,12 @@ def upload_pipeline(
         logger.info("Pipeline uploaded")
         return pipeline
 
-    def delete_pipeline(self, pipeline_id):
```
Comment (roytman) [inline kfp/kfp_support_lib/kfp_v2_workflow_support/src/workflow_support/pipeline_utils/pipeline_utils.py:75]: we don't have `delete_pipeline` now?

File: .github/workflows/test.yml (hunk 3 above)
Comment (revit13) [reply .github/workflows/test.yml]: @roytman I change the steps description, Thanks

File: kfp/kfp_support_lib/kfp_v2_workflow_support/src/workflow_support/pipeline_utils/pipelines_tests_utils.py [hunk 7]
```diff
@@ -25,16 +25,17 @@ def run_test(pipeline_package_path: str, endpoint: str = "http://localhost:8080/
...
         pipeline_name=pipeline_name,
-        overwrite=overwrite,
     )
     if pipeline is None:
         return None
     experiment = utils.get_experiment_by_name()
-    run_id = utils.start_pipeline(pipeline, experiment, params=[])
-    status, error = utils.wait_pipeline_completion(run_id=run_id, timeout=tmout, wait=wait)
-    if status.lower() not in ["succeeded", "completed"]:
+    run_id = utils.start_pipeline(pipeline, experiment, params={})
```
Comment (roytman) [inline kfp/kfp_support_lib/kfp_v2_workflow_support/src/workflow_support/pipeline_utils/pipelines_tests_utils.py:32]: can it be `assert run_id is not None` ?

File: kfp/superworkflows/Makefile [hunk 8]
```diff
@@ -18,12 +18,9 @@ workflow-test:
 workflow-upload:
 	$(MAKE) -C ray/kfp_v1 workflow-upload
 
-.PHONY: workflow-reconcile-requirements
-workflow-reconcile-requirements:
-	$(MAKE) -C ray/kfp_v1 workflow-reconcile-requirements
-
-set-versions:  #workflow-reconcile-requirements
-	@# Help: Nothing to do here yet.
+.PHONY: set-versions
```
Comment (roytman) [inline kfp/superworkflows/Makefile:40]: should it be executed if KFPv2 is not 1 ?

File: transforms/language/lang_id/kfp_ray/lang_id_wf.py [hunk 9]
```diff
@@ -104,9 +104,9 @@ def lang_id(
     # Ray cluster
     ray_name: str = "lang_id-kfp-ray",  # name of Ray cluster
     ray_head_options: str = '{"cpu": 1, "memory": 4, "image_pull_secret": "", '
-    '"image": "' + task_image + '", "image_pull_policy": "Always" }',
+    '"image": "' + task_image + '", "image_pull_policy": "" }',
     ray_worker_options: str = '{"replicas": 2, "max_replicas": 2, "min_replicas": 2, "cpu": 2, "memory": 4, '
-    '"image_pull_secret": "", "image": "' + task_image + '", "image_pull_policy": "Always" }',
+    '"image_pull_secret": "", "image": "' + task_image + '", "image_pull_policy": "" }',
```
Comment (roytman) [inline transforms/language/lang_id/kfp_ray/lang_id_wf.py]: if you don't set `image_pull_policy` why don't you remove it at all?

File: transforms/universal/noop/kfp_ray/noop_multiple_wf.py [hunk 10]
```diff
@@ -97,9 +97,9 @@ def noop(
     # Ray cluster
     ray_name: str = "noop-kfp-ray",  # name of Ray cluster
     ray_head_options: str = '{"cpu": 1, "memory": 4, "image_pull_secret": "", '
-    '"image": "' + task_image + '", "image_pull_policy": "Always" }',
+    '"image": "' + task_image + '", "image_pull_policy": "" }',
```
Comment (roytman) [inline transforms/universal/noop/kfp_ray/noop_multiple_wf.py]: the same comment about `image_pull_policy`

File: transforms/universal/tokenization/ray/Makefile [hunk 11]
```diff
@@ -33,6 +33,10 @@ setup:: .transforms.setup
 # distribution versions is the same as image version.
 set-versions:
 	$(MAKE) TOML_VERSION=$(DOCKER_IMAGE_VERSION) .defaults.update-toml
+	cat pyproject.toml | sed                                                                   \
```
Comment (roytman) [inline transforms/universal/tokenization/ray/Makefile:36]: I'm not sure that we need it

**Review by roytman — APPROVED** [review]
LGTM

**Pull request merged**